src/main/java/bio/terra/cli/service/WorkspaceManager.java

```diff
@@ -228,89 +245,224 @@ public RoleBindingList listUsersOfWorkspace() {
         .getRoles(workspaceContext.getWorkspaceId());
   }
 
+  // ====================================================
+  // Workspace resources, controlled & referenced
+
   /**
-   * Lookup a controlled resource by its name. Names are unique within a workspace.
-   *
-   * @param resourceName name of resource to lookup
-   * @return the cloud resource object
-   * @throws UserActionableException if the resource is not controlled (e.g. external bucket)
+   * Update the cached list of resources (controlled & referenced) in the workspace, by fetching the
+   * up-to-date list from WSM.
    */
-  public CloudResource getControlledResource(String resourceName) {
+  private void updateResourcesCache() {
+    // check that there is a workspace currently mounted
```

Comment:
no action: I was thinking we might want to have a command for updating the cache, but I guess as long as listing and resolving resources does that we're ok. Maybe worth documenting for users somewhere so that if their collaborator changes something about a workspace they know they need to do something before calling docker run to get a new resource env var. I can imagine it being very confusing to have resources changed out from under you. Not sure what the best way to do that is.

Reply:
Agreed about adding this to the documentation. I'm planning to update the README in a follow-on ticket, after I get everyone's okay on this command reorganization, in particular.